omod/src/main/java/org/openmrs/module/pihcore/page/controller/PatientPageController.java

```diff
@@ -0,0 +1,62 @@
+package org.openmrs.module.pihcore.page.controller;
+
+import org.openmrs.Patient;
+import org.openmrs.PatientIdentifier;
+import org.openmrs.PatientIdentifierType;
+import org.openmrs.api.PatientService;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.converter.util.ConversionUtil;
+import org.openmrs.ui.framework.page.PageModel;
+import org.openmrs.ui.framework.page.Redirect;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class PatientPageController {
```

Comment:
Might be good to add a comment as to where/why we are using this, as years down the road we will probably forget.

Also, I'm probably missing something obvious, but I'm not 100% why this is needed at all? Is the full patient object not available in the Core Apps context where you render the link (that is, could the link just have a reference to the patient uuid)?

Reply:
Thanks @mogoodrich - comment added.  Per the comment, the reason this is needed is because we are not linking to a page in the same EMR instance, so the uuid and primary key id of the patient you need to link to is unknown.  The shared information that links the patient across the 2 systems is the HIVEMR-v1 identifier that is present in both.

Reply:
Duh, right, thanks @mseaton !